compiler/optimizer/CopyPropagation.cpp

```diff
@@ -540,18 +544,32 @@ int32_t TR_CopyPropagation::perform()
                            prevDefNode,
                            prevDefNode->getDecimalPrecision());
 
+                        auto equivalentDefLookup = equivalentDefs.find(j);
                         if ((!thisSymRefIsPreferred &&
-                             (prevSymRefIsPreferred || (equivalentDefs[j-firstRealDefIndex] != j))))
+                             (prevSymRefIsPreferred || (equivalentDefLookup != equivalentDefs.end()))))
                            {
                            if (trace())
                               traceMsg(comp(), "000setting i %d to j %d\n", i, j);
-                           equivalentDefs[i-firstRealDefIndex] = equivalentDefs[j-firstRealDefIndex];
+                           if (equivalentDefLookup == equivalentDefs.end())
+                              {
+                              if (i != j)
+                                 equivalentDefs[i] = j;
+                              }
+                           else if (i == equivalentDefLookup->second)
+                              {
+                              equivalentDefs.erase(i);
+                              }
+                           else
+                              {
+                              equivalentDefs[i] = equivalentDefLookup->second;
+                              }
                            }
-                        else if (equivalentDefs[j-firstRealDefIndex] == j)
+                        else if (equivalentDefs.find(j) == equivalentDefs.end())
                            {
                            if (trace())
                                traceMsg(comp(), "111setting j %d to i %d\n", j, i);
-                           equivalentDefs[j-firstRealDefIndex] = i;
+                           if (i != j)
+                              equivalentDefs[j] = i;
                            }
```

Comment:
Is there a reason there is'nt logic similar to what I saw in an earlier instance of this kind of condition in this commit : 

else if (i == equivalentDefLookup->second)
                              {
                              equivalentDefs.erase(i);
                              }

that I assume is essentially saying that if we are just marking a def as it's own equivalent def, then we do not need an entry for it in the map data structure that we have now.